R/EnrichDepletTF.R

```diff
@@ -0,0 +1,417 @@
+#'
+
+sysGTRDgenes <- function(GeneSym){
+
+  GTRDgenes <- fetchData("GTRDgeneTFs")
+  GTRDgenes <- GTRDgenes[which(names(GTRDgenes) %in% GeneSym)] #Keeping the data for
+                                                               #the specified genes
+  return(GTRDgenes)
+}
+
+# ====== 1.3  Loading the GTRDTFgenes data for chosen genes and TF =============================
+
+#' sysGTRDtf.
+#'
+#' \code{sysGTRDtf} Fetch GTRD data for the specified Transcription Factors and genes.
+#'
+#' @section Details: The fetched dataset was collected and composed by Boris Steipe
+#'\href{https://github.com/hyginn/BCB420.2019.ESA/blob/master/R/fetchData.R{fetchData}.
+#' The GTRDTFgenes dataset lists which genes bind each specified TF, as observed by
+#' ChIP-seq experiments that compose the GTRD database. The loading function was written
+#' by Boris Steipe
+#'\href{https://github.com/hyginn/BCB420.2019.ESA/blob/master/R/fetchData.R{fetchData}.
+#'
+#' @param TfSym (char vector)   A character vector of length > 0L of unique TFs
+#'                              (HGNC symbols) that was extracted from the output
+#'                              of sysGTRDgenes() for a certain set of genes.
+#' @param GeneSym (char vector) A character vector of length > 0L of the unique genes
+#'                              (HGNC symbols) that the function sysGTRDgenes()
+#'                              retrieved for a certain set of genes (the same certain
+#'                              set of genes that was used to produce TfSym).
+#' @return (list)               A list of transcription-factor-named character vectors of
+#'                              the genes that bind each transcription factor.
+#'
+#' @family <optional description of family>
+#'
+#' @author \href{https://orcid.org/0000-0002-9478-5974}{Sapir Labes} (aut)
+#'
+#' @seealso \code{\link{<function>}} <describe related function>, ... .
+#'
+#' @examples
+#' # Fetching the vector of the genes that bind the transcription factor AR.
+#' GTRDgenes <- sysGTRDgenes(GeneSym = c("LAMP1", "HDAC6"))
+#' GeneSym <- names(GTRDgenes) #The specific set of genes retrieved
+#' TfSym <- unique(unlist(GTRDgenes, use.names = FALSE)) #TFs that bind the specific set of genes
+#' sysGTRDtf(TfSym = TfSym, GeneSym = GeneSym)
+#'
+
+sysGTRDtf <- function(TfSym, GeneSym){
+
+  GTRDtf <- fetchData("GTRDTFgenes")
+  GTRDtf <- GTRDtf[which(names(GTRDtf) %in% TfSym)] #keeps only TFs that appear
+                                                    #in TfSym (i.e., TF that binds
+                                                    #the chosen genes.
+  GTRDtf <- lapply(GTRDtf, function(x) x[which(x %in% GeneSym)]) #Keep only genes
+                                                                 #that are in GeneSym
+
+  #Check:
+  if (identical(sort(unique(unlist(GTRDtf, use.names = FALSE))) ,sort(GeneSym))){
+    return(GTRDtf)
+  }else{
+    return(print("Error: input genes are not equal to output genes"))
```

Comment:
Definitely don't return a string that is semantically different from the output. And what you are actually returning is the return-value of `print()`. Use `stop("<message>") instead.